### PR TITLE
Remove primary ctor when moving a type.

### LINKED ISSUE #72632: Move type code fix results in compile error with primary constructors
**Given:**

```
internal class Outer()
{
    private class Inner$$
    {
    }
}
```

With cursor at $$, run the "Move type to Outer.Inner.cs" code action.

**Expected:**

```
internal partial class Outer()
{
}
```

and

```
internal partial class Outer
{
    private class Inner
    {
    }
}
```

**Actual:**

```
internal partial class Outer()
{
}
```

and

```
internal partial class Outer()
{
    private class Inner
    {
    }
}
```

The parenthesis being on both `Outer` declarations is a compile error.

### PR DESCRIPTION
**Pull request by juan-carlos-diaz**
Fix #72632

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs [hunk 1]
```diff
@@ -1200,6 +1200,17 @@ private static SyntaxNode WithAttributeLists(SyntaxNode declaration, SyntaxList<
             _ => declaration,
         };
 
+    internal override ImmutableArray<SyntaxNode> GetPrimaryCtor(SyntaxNode declaration)
+    {
+        if (declaration is ClassDeclarationSyntax || declaration is StructDeclarationSyntax)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs:1205]: can just check for `TypeDeclarationSyntax`

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs [hunk 2]
```diff
@@ -1200,6 +1200,17 @@ private static SyntaxNode WithAttributeLists(SyntaxNode declaration, SyntaxList<
             _ => declaration,
         };
 
+    internal override ImmutableArray<SyntaxNode> GetPrimaryCtor(SyntaxNode declaration)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs:1203]: not sure why this is returning an array.

Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs:1203]: GetPrimaryConstructorParameterList

**Comment by CyrusNajmabadi** [comment]
Done with pass.

File: src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeEditor.cs [hunk 3]
```diff
@@ -125,7 +125,7 @@ private async Task<Document> AddNewDocumentWithSingleTypeDeclarationAsync(Docume
             // attributes from the containing partial types.  We don't want to create
             // duplicate attributes on things.
             AddPartialModifiersToTypeChain(
-                documentEditor, removeAttributesAndComments: true, removeTypeInheritance: true);
+                documentEditor, removeAttributesAndComments: true, removeTypeInheritance: true, removePrimaryCtor: true);
```
[the remark this answers is not in the document]
Comment (juan-carlos-diaz) [reply src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.MoveTypeEditor.cs]: Done

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs (hunk 1 above)
Comment (juan-carlos-diaz) [reply src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs]: Done

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs (hunk 2 above)
Comment (juan-carlos-diaz) [reply src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs]: Done

Comment (juan-carlos-diaz) [reply src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs]: Change to "SyntaxNode?"

File: src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs [hunk 4]
```diff
@@ -1200,6 +1200,17 @@ private static SyntaxNode WithAttributeLists(SyntaxNode declaration, SyntaxList<
...
+    internal override SyntaxNode? GetPrimaryConstructorParameterList(SyntaxNode declaration)
+    {
+        if (declaration is TypeDeclarationSyntax)
+        {
+            var paramList = declaration.GetParameterList();
+            if (paramList is not null)
+                return paramList;
+        }
+        return null;
+    }
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs:1212]:
```suggestion
    internal override SyntaxNode? GetPrimaryConstructorParameterList(SyntaxNode declaration)
        => declaration is TypeDeclarationSyntax { ParameterList: { } parameterList } ? parameterList : null;
```

Comment (juan-carlos-diaz) [reply src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs]: Beautiful!!! Thanks

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**